[wip] asim: add randomness to range generation

wenyihu6 opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106586)
<!-- Reviewable:end -->

wenyihu6 committed:
asim: introduce rand testing framework with default-only settings

This patch lays the backbone of the randomized testing framework. Currently, it
only supports default configuration for all options, implying that there is no
randomization yet. Additionally, it refactors some of the existing structure in
data_driven_test. Note that this should not change any existing behavior, and
the main purpose is to make future commits cleaner.

Part of: https://github.com/cockroachdb/cockroach/issues/106311
Release note: None

wenyihu6 committed:
addrand

wenyihu6 committed:
asim: allow rand testing to pick a random predefined configuration

This patch takes the first step towards a randomized framework by enabling asim
testing to randomly select a cluster information configuration from a set of
predefined choices. These choices are hardcoded and represent common cluster
configurations.

Part of: https://github.com/cockroachdb/cockroach/issues/106311
Release note: None
